src/assisted_test_infra/test_infra/helper_classes/config/base_day2_cluster_config.py

```diff
@@ -14,7 +14,8 @@ class BaseDay2ClusterConfig(BaseClusterConfig, ABC):
     day1_cluster_details: Optional[models.cluster.Cluster] = None
     day1_base_cluster_domain: Optional[str] = None
     day1_api_vip_dnsname: Optional[str] = None
-    day2_workers_count: Optional[int] = None
+    day2_workers_count: int = None
+    day2_masters_count: int = None
```

Comment:
```suggestion
    day2_workers_count: Optional[int] = None
    day2_masters_count: Optional[int] = None
```
The variable is `Optional` as long as its value can be `None`

Reply:
Yes when its Optional  , the param set fails because base_test.py def update_parameterized(cls, request: pytest.FixtureRequest, config: BaseConfig): 
The failure happens here : def _get_correct_value(self, attr: str, new_val):  in base_config.py

      ```
  if hasattr(_type, "__origin__"):
            return _type.__origin__(new_val)  ------------> code goes here and reutrn raise error

        # str, int, float, bool, Path, and more
        return new_val if isinstance(new_val, _type) else _type(new_val)

```
 ```
@pytest.mark.parametrize("day2_masters_count", [1])
@pytest.mark.parametrize("day2_workers_count", [1])
```

Reply:
@eliorerz any idea why it fails?

Reply:
```
(Pdb++) ll
  59         def _get_correct_value(self, attr: str, new_val):                                                                                                                                                                                 
  60             """Get value in its correct type"""                                                                                                                                                                                           
  61             pdb.set_trace()                                                                                                                                                                                                               
  62             annotations = self.get_annotations()                                                                                                                                                                                          
  63             if not hasattr(self, attr):                                                                                                                                                                                                   
  64                 raise AttributeError(f"Can't find {attr} among {annotations}")                                                                                                                                                            
  65                                                                                                                                                                                                                                           
  66  ->         _type = annotations[attr]                                                                                                                                                                                                     
  67                                                                                                                                                                                                                                           
  68             if hasattr(_type, "__origin__"):                                                                                                                                                                                              
  69                 return _type.__origin__(new_val)                                                                                                                                                                                          
  70                                                                                                                                                                                                                                           
  71             # str, int, float, bool, Path, and more                                                                                                                                                                                       
  72             return new_val if isinstance(new_val, _type) else _type(new_val)                                                                                                                                                              
(Pdb++) n
[44] > /home/benny/assisted-test-infra/src/assisted_test_infra/test_infra/helper_classes/config/base_config.py(68)_get_correct_value()
-> if hasattr(_type, "__origin__"):
   5 frames hidden (try 'help hidden_frames')
(Pdb++) pp _type
typing.Optional[int]
(Pdb++) p attr
'day2_workers_count'
(Pdb++) pp _type.__dict__
{'__args__': (<class 'int'>, <class 'NoneType'>), '__origin__': typing.Union, '__parameters__': (), '__slots__': None, '_inst': True, '_name': 'Optional', '_paramspec_tvars': False}
(Pdb++) p new_val
1
(Pdb++) _type.__origin__(new_val)
*** TypeError: Cannot instantiate typing.Union
Traceback (most recent call last):
  File "/root/.pyenv/versions/3.11.0/lib/python3.11/typing.py", line 463, in __call__
    raise TypeError(f"Cannot instantiate {self!r}")
(Pdb++) 

```

Reply:
I think that it shouldn't be `Optional` type.
We use those annotations to cast some arguments during the initiation process

Reply:
I'm working on a fix to support Optional in `parametrize`

Reply:
ack ,  its not something that impact on the current test ?

Reply:
https://github.com/openshift/assisted-test-infra/pull/2184